Split into backend and frontend

xmnlab opened the pull request.

saulshanabrook commented on backend/README.md:
````diff
@@ -0,0 +1,63 @@
+
+```bash
+
+npm install
````
Not a big deal, but we use [`yarn`](https://yarnpkg.com/en/) in JupyterLab. the `jlpm` command is really just an alias for `yarn`. It is an alternative to `npm`. 

I think using yarn on the backend server makes sense, instead of npm.

xmnlab replied:
I didn't know. thanks for the explanation. I will change to yarn :)

saulshanabrook commented on backend/README.md:
```diff
@@ -0,0 +1,63 @@
+
+### Build and Run the Tests
+
+To build this extesion run:
```
Do we have a build step for the backend?

xmnlab replied:
actually it should be for the backend .. i will fix the text.

xmnlab replied:
or just start would be enough?

saulshanabrook replied:
Yeah I think the start is enough. I haven't tried it though

xmnlab commented on README.md:
````diff
@@ -15,41 +15,100 @@ A metadata service, client and UI for the metadata schema of [schema.org](https:
+jlpm run build
+```
+
+Edit the files in the package in `./packages/jupterlab-commenting` and run `jupyter lab --watch --dev` in the 
````
not sure if this instructions should be here @saulshanabrook any thoughts?

saulshanabrook replied:
I put the wrong folder here. I fixed it below.

xmnlab commented on README.md:
```diff
@@ -15,41 +15,100 @@ A metadata service, client and UI for the metadata schema of [schema.org](https:
+
+TODO: Write instructions for how to contribute to the backend server (e.g. how to run in development mode).
+
+#### Frontend Extension
```
These instructions didn't work for me. maybe it is missing extra steps.

xmnlab commented on README.md:
```diff
@@ -15,41 +15,100 @@ A metadata service, client and UI for the metadata schema of [schema.org](https:
+
+
+##### JupyterLab master
+To develop against an unreleased version of `JupyterLab` requires that you add this extension
```
These instructions didn't work for me. maybe it is missing extra steps.

xmnlab commented on README.md:
```diff
@@ -15,41 +15,100 @@ A metadata service, client and UI for the metadata schema of [schema.org](https:
+jlpm run add:sibling ../jupyterlab-metadata-service/frontend
+cd packages/frontend
+# build metadata frontend
+jlpm run build
```
as we (@hoo761 ) were talking today, maybe this command should be run on the jupyterlab root folder (at least for the first time) to build also jupyterlab because without that it will raise errors.

saulshanabrook replied:
Yeah it sounds reasonable to build jupyterlab core itself before running `add:sibling`

saulshanabrook commented on README.md:
````diff
@@ -15,41 +15,100 @@ A metadata service, client and UI for the metadata schema of [schema.org](https:
+jlpm run build
+```
+
+Edit the files in the package in `./packages/jupterlab-commenting` and run `jupyter lab --watch --dev` in the 
````
```suggestion
Edit the files in the package in `./packages/frontend` and run `jupyter lab --watch --dev` in the 
```

xmnlab commented:
@saulshanabrook @acu192 

I added a fixed connection with port 40000. 
I added a fixed mutation using data from a variable.

for some unknown reason the query just get the first data ... on graphql playground I can see all the items.  I will investigate this problem.
